constellations/orion/src/controllers/classes_by_semesters_controller_test.go

```diff
@@ -0,0 +1,223 @@
+package controllers_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/controllers/testUtils"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOneSemesterOneProgramOneClass_Success(t *testing.T) {
+	// Mock 1 program, 1 semester, 1 class
+	testUtils.ProgramRepo.MockSelectAll = func(publishedOnly bool) ([]domains.Program, error) {
```

Comment:
This is how you define a function to return two things: a list of Programs, and an error. If no error (right below), it is nil.